### PR TITLE
Add Xlift AMP support

### PR DESCRIPTION
**Pull request by interspacejp**
Could you merge X-Lift(native ad) amp-ad support into your master branch?
This request will add a "xlift" type to amp-ad.

We have already signed a Google CLA.

Corporation Name: Interspace Co.,Ltd
Thank you.

### CODE REVIEW

**Comment by zhouyx** [comment]
@interspacejp Thanks for the Pull Request. 
We have a [renderStart API](https://github.com/ampproject/amphtml/blob/master/ads/README.md#available-apis) that helps AMP to provide better ad loading experience. 
Could you take a look and implement it (renderStart + noContentAvailable)?
Let me know if there's any question.

**Comment by interspacejp** [comment]
@zhouyx Thanks for your review.
I implemented your request and changed something.
but i can't pass travis-ci test.

File: ads/xlift.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+</amp-ad>
+```
+
+## Configuration
+
+For configuration details and to generate your tags, please contact https://www.x-lift.jp/#contact
+
+Supported parameters:
+
+- width
```
Comment (zhouyx) [inline ads/xlift.md]: `width` and `height` are required for any amp-ad. You don't need to include them in your doc. Also you may want to indicate that `data-mediaid` is required.

**Comment by zhouyx** [comment]
travis-ci test failed is fixed now.

File: ads/xlift.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  d.addEventListener('SuccessLoadedXliftAd', function(e) {
+    e.detail = e.detail || {adSizeInfo: {}};
+    global.context.renderStart(e.detail.adSizeInfo);
+  });
+  d.addEventListener('FailureLoadedXliftAd', function() {
+    global.context.noContentAvailable();
+  });
+
+  loadScript(global, 'https://cdn.x-lift.jp/resources/common/xlift_amp.js', () => {
+    global.XliftAmpHelper = global.XliftAmpHelper || null;
```
Comment (interspacejp) [inline ads/xlift.js]: I think so . But travis ci can't recognized XmlAmpHelper if remove this line.
For reason,  I added new line for assign XmlAmpHelper property to global before executed loadScript.

**Comment by interspacejp** [comment]
Thank you for your opinion. I changed something again. 
Please check it.

**Comment by zhouyx** [comment]
Looks great! Ping @lannka to take a final look

**Comment by interspacejp** [comment]
@zhouyx  Thanks! and i want to a question.
What dose  "Use upgrade callback to pick the correct carousel class. #5899" from @camelburrito mean?
@lannka Please check my changes.Thanks.

**Comment by zhouyx** [comment]
@interspacejp Thanks for the Pull Request. Merged.
@camelburrito You want to explain #5899 a bit?

**Pull request merged**